Dropout.lua

```diff
@@ -20,13 +21,14 @@ function Dropout:updateOutput(input)
       self.output:resizeAs(input):copy(input)
    end
    if self.p > 0 then
-      if self.train then
-         self.noise:resizeAs(input)
-         self.noise:bernoulli(1-self.p)
+      local si = self.si and not self.train
+      if self.train or si and not self.v2 then
+         local noise = si and input:clone() or self.noise:resizeAs(input)
```

Comment:
Thanks for pointing this out. I will remove the cloning and use just `self.noise`. Has the feature of sharing `noise` field among layers been implemented in your [optnet](https://github.com/fmassa/optimize-net) library?